app/models/wallet-connect-client/wallet-connect-client.ts

```diff
@@ -0,0 +1,247 @@
+      error: Error | null,
+      payload: any | null,
+    ) {
+      if (error) {
+        throw error
+      }
+
+      const { peerMeta, peerId } = self.connector
+      console.log("handleNewCallRequest", payload)
+      self.navigationStore.navigateTo({
+        routeName: "App",
+        action: NavigationActions.navigate({
+          routeName: "WalletConnect",
+          params: {
+            peerId,
+            peerMeta,
+            payload,
+          },
+        }),
+      })
+    },
+    handleDisconnect(error: Error | null) {
+      if (error) {
+        throw error
+      }
+      self.serializedSession = ""
+    },
+  }))
+  .actions(self => ({
+    createSession(uri: string) {
+      self.connect({ uri })
+      self.connector.on("session_request", self.handleNewSessionRequest)
+    },
+    restoreSession() {
+      const session: WalletConnectSession = JSON.parse(self.serializedSession)
+      self.connect({ session })
+
+      self.connector.on("call_request", self.handleNewCallRequest)
+      self.connector.on("disconnect", self.handleDisconnect)
+    },
+    approveSession() {
+      self.connector.approveSession({
+        chainId: 99999,
+        accounts: [],
+      })
+      self.connector.off("session_request")
+
+      self.serializedSession = JSON.stringify(self.connector.session)
+
+      self.connector.on("call_request", self.handleNewCallRequest)
+      self.connector.on("disconnect", self.handleDisconnect)
+    },
+    rejectSession() {
+      self.connector.rejectSession()
+    },
+    approveCallRequest(response: {
+      id: number
+      result: any
+    }) {
+      self.connector.approveRequest(response)
+    },
+    rejectCallRequest(response: {
+      id: number
+      error: any
+    }) {
+      self.connector.rejectRequest(response)
+    },
+  }))
+  .actions(self => ({
+    rejectRequest(payload: any) {
+      if (isSessionRequest(payload.method)) {
+        self.rejectSession()
+      } else {
+        self.rejectCallRequest({
+          id: payload.id,
+          error: { message: "User rejected call request" }
+        })
+      }
+    },
+  }))
+  .actions(self => ({
+    handleCallRequestApproval: flow(function * (payload: any) {
+      let result = null
+      try {
+        switch (payload.method) {
+          case "keplr_enable_wallet_connect_v1":
+            result = []
+            break
+  
+          case "keplr_get_key_wallet_connect_v1":    
+            result = [
+              yield self.env.authCoreAPI.getWalletConnectGetKeyResponse(
+                payload.params[0],
+                { name: self.currentUser.likerID }
+              ),
+            ]
+            break
+  
+          case "keplr_sign_amino_wallet_connect_v1":
+            try {
+              result = yield self.env.authCoreAPI.signAmino(
+                payload.params[2],
+                payload.params[1]
+              )
+            } catch (error) {
+            }
+            break
+  
```

Comment:
Please wrap the cases in `{}`s, and comment what do the `params` mean in each case if possible